java/arcs/core/data/proto/ParticleProtoDecoder.kt

```diff
@@ -38,7 +38,13 @@ fun HandleConnectionProto.decode(
     val recipeHandle = requireNotNull(context.recipeHandles[handle]) {
         "Handle '$handle' not found when decoding ParticleProto '${particleSpec.name}'."
     }
-    return HandleConnection(handleSpec, recipeHandle)
+    val connectionType = try { type.decode() } catch (e: IllegalArgumentException) {
+        throw IllegalArgumentException(
```

Comment:
Why do we need to do this? When would a handle connection type be missing?

Reply:
It shouldn't be missing within normal usage. An error may be thrown when a developer writes or updates a test or is creating new functionality. In this case, error messaging is friendlier. I'm trying to program defensively. Ideally, the above cases wouldn't have nulls, but they will in exceptional cases. Here, we provide helpful messages. This is the line of thinking behind this exception handling logic.

Reply:
If we add rethrows with nicer messages for every field in the proto that may be missing (and everything may be missing in protos, everything is optional!) our code will grow 3-5x in size (LOC) while not providing benefit as exceptions already have stack traces that will tell you exactly where the exception happened. I would advise against adding such bloat.